packages/core/src/components/DateField/DateField.jsx

```diff
@@ -268,6 +271,10 @@ DateField.propTypes = {
    * Label for the year `input` field
    */
   yearLabel: PropTypes.node,
+  /**
+   * Maximum number of digits to allow in the year `input` field
+   */
+  yearLimit: PropTypes.number,
```

Comment:
I'm not sure if I understand the scenario where we would want a different number than 4. My understanding of this component is that it only allows `DD/MM/YYYY`(although maybe we want to support `DD/MM/YY` in the future?)

Reply:
Yes, I included this new property to allow for two-digit years as an option. We predominantly use four-digit years in MCT, but there are a few places where we prefer two-digit years instead.

Reply:
I think it makes sense for that functionality to be done in a different PR because it sounds like we should also style the year field to be shorter.